src/guidellm/executor/profile_generator.py

```diff
@@ -61,25 +61,24 @@ def next_profile(
         pass
 
 
-@ProfileGenerator.register_generator(ProfileGenerationModes.SINGLE)
-class SingleProfileGenerator(ProfileGenerator):
-    def __init__(self, rate: float, rate_type: str, **kwargs):
-        super().__init__(ProfileGenerationModes.SINGLE)
-        self._rate = rate
+@ProfileGenerator.register_generator(ProfileGenerationModes.MULTI)
+class MultiProfileGenerator(ProfileGenerator):
+    def __init__(self, rate: List[float], rate_type: str, **kwargs):
+        super().__init__(ProfileGenerationModes.MULTI)
+        self._rates = rate
+        self._rate_index = 0
         self._rate_type = rate_type
         self._generated = False
 
     def next_profile(
         self, current_report: TextGenerationBenchmarkReport
     ) -> Optional[Profile]:
-        if self._generated:
+        if self._rate_index >= len(self._rates):
             return None
 
-        self._generated = True
-
         if self._rate_type == "constant":
```

Comment:
Can we simplify the constant and poisson since the only thing changing between these now is the load_gen_mode?

Reply:
I think that was already the case between these two (they differed in load_gen_mode and had the same load_gen_rate as they still do), this is an attempt but I can't say I think it is definitely an improvement:
https://github.com/neuralmagic/guidellm/pull/2/commits/eb03706bb48e1a815405a4979777d8d66b223a98